lib/sol/SolRpc.js

```diff
@@ -1,322 +1,454 @@
-   * @param {number} [config.port] - The port number (optional).
-   * @param {string} config.account - The account to use for transactions.
-   */
+ * Constructs a new instance of the SolRPC class.
+ * 
+ * @param {Object} config - The configuration object containing the connection details.
+ * @param {string} config.protocol - The network protocol (e.g., 'http', 'https', 'wss').
+ * @param {string} config.host - The host URL or IP address.
+ * @param {number} [config.port] - The port number (optional).
+ * @param {string} config.account - The account to use for transactions.
+ */
   constructor(config) {
     this.config = config;
-    this.connection = this.initConnection(this.config);
+    this.rpcUrl = `${this.config.protocol}://${this.config.host}`;
+    this.rpcSubscriptionUrl = this.rpcUrl.replace('http', 'ws');
+    this.rpc = this.initRpcConnection(this.config);
+    this.rpcSubscriptions = this.initRpcSubscriptions(this.config);
     this.emitter = new EventEmitter();
-    this.web3 = Web3;
     // configuration for retrieving versioned blocks and transactions
     this._versionedConfig = {
       commitment: 'confirmed',
       maxSupportedTransactionVersion: 0
     };
+
+
   }
 
   /**
-   * Creates and returns a new Web3 Connection instance configured with the specified connection settings.
-   * The connection is set to return data that has been confirmed by the cluster.
-   * 
-   * @param {Object} connectionConfig - The configuration object containing the connection details.
-   * @param {string} connectionConfig.protocol - The network protocol (e.g., 'http', 'https', 'wss').
-   * @param {string} connectionConfig.host - The host URL or IP address.
-   * @param {number} [connectionConfig.port] - The port number (optional).
-   * @throws {Error} If the protocol is not specified or is invalid.
-   * @returns {Web3.Connection} A new Web3 Connection instance.
-   */
-  initConnection(connectionConfig) {
-    const { protocol, host, port } = connectionConfig;
-    if (!protocol || !['wss', 'http', 'https'].includes(protocol.toLowerCase())) {
-      throw new Error('Please provide a valid protocol');
+ * Creates and returns a new Web3 Connection instance configured with the specified connection settings.
+ * The connection is set to return data that has been confirmed by the cluster.
+ * 
+ * @returns {import('@solana/web3.js').Rpc<import('@solana/web3.js').SolanaRpcApi>}
+ */
+  initRpcConnection() {
+    const rpc = createSolanaRpc(this.rpcUrl);
+    return rpc;
+  }
+
+  initRpcSubscriptions() {
+    return createSolanaRpcSubscriptions(this.rpcSubscriptionUrl);
+  }
+
+  async cleanup() {
+    // Close WebSocket connections and remove subscriptions
+    if (this.rpcSubscriptions) {
+      // Remove all subscription listeners first
+      const subscriptionEvents = this.rpcSubscriptions.eventNames();
+      if (typeof subscriptionEvents === 'object') {
+        Object.keys(subscriptionEvents).forEach(event => {
```

Comment:
Don't use .forEach. Ever. `for` loops are faster, easier to debug, and throw immediately.